### PR TITLE
Added AdvancedWebView

### PR DESCRIPTION
**Pull request by ArjixWasTaken**
This is a webview that can be controlled though predefined actions/conditions, it attempts to mimic the selenium API.

~~This is still in alpha, it may be prone to bugs.~~
This is now in beta, it may still be prone to bugs but it is more stable.

Current known bugs:
- (Possibly fixed) ~~Sometimes an action is ran twice for no reason, might have to do with the concurrency~~.

### CODE REVIEW

**Comment by ArjixWasTaken** [comment]
I can't find any real issue right now, might be better to make a provider that uses this WebView helper in order to test drive it.

File: app/src/main/java/com/lagradost/cloudstream3/network/WebViewResolver.kt [hunk 1]
```diff
@@ -14,8 +16,346 @@ import kotlinx.coroutines.runBlocking
...
+    private var isInSleep = false
+    private var actionStartTimestamp = -1;
+
+    var Error = "None"
+
+    private suspend fun tryExecuteAction() {
+        if (actionExecutionsPaused || remainingActions.size == 0) return
+        actionExecutionsPaused = true
+        actionStartTimestamp = (System.currentTimeMillis() / 1000).toInt()
+        setCurrentHTML()
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/network/WebViewResolver.kt:114]: impo you should instead do smth like setCurrentHTML { Instance.run(action.callback) } so you can be 100% sure that the html is up to date and only updated when actually used. As setCurrentHTML is not guaranteed to be set immediately on called. I DONT WANT ANY SORT OF RACE CONDITION POSSIBLE

setCurrentHTML(callback : (() -> Unit)) {
webView?.evaluateJavascript("document.documentElement.outerHTML")?.let { currenthtml =... 
callback.invoke()
  }
}

Comment (LagradOst) [reply app/src/main/java/com/lagradost/cloudstream3/network/WebViewResolver.kt:114]: you can abstract it away to only take the action as an argument and run, so setHtmlAndRun(action.callback)

Comment (ArjixWasTaken) [reply app/src/main/java/com/lagradost/cloudstream3/network/WebViewResolver.kt]: I think this is now resolved?

**Comment by LagradOst** [comment]
LGTM, @Blatzar thoughts?

**Comment by Blatzar** [comment]
:shrug: are all the actions working

**Comment by ArjixWasTaken** [comment]
> :shrug: are all the actions working

When tested individually? Yes

As I mentioned above a provider that uses this WebView would easily show us any remaining flaws/bugs.

**Comment by LagradOst** [comment]
LGTM, ready to merge? 👍

**Comment by ArjixWasTaken** [comment]
> LGTM, ready to merge? 👍

Let me make some more commits before that.

**Comment by ArjixWasTaken** [comment]
Ok I'll make this a draft once again, gotta make some QOL fixes.

**Pull request closed without merging**